test/structureHash.test.ts

```diff
@@ -0,0 +1,50 @@
+import fetch from "jest-fetch-mock";
+import { connect } from "../src/main";
+import {
+  FetchHashTableSuccess,
+  FetchHashTableError
+} from "./fauxFetch";
+
+describe("has method", function () {
+  let connection: any;
+  beforeAll(async function () {
+    // reset in case another test file hasn't cleaned up
+    fetch.resetMocks();
+    connection = await connect({
+      network: "testnet",
+      host: "https://testnet.tableland.network",
+    });
+  });
+
+  afterEach(function () {
+    // ensure mocks don't bleed into other tests
+    fetch.resetMocks();
+  });
+
+  test("Hashing a table works", async function () {
+    fetch.mockResponseOnce(FetchHashTableSuccess);
+
+    const createStatement = "CREATE TABLE hello (id int primary key, val text);";
+    const hashResponse = await connection.hash(createStatement);
+
+    const payload = JSON.parse(fetch.mock.calls[0][1]?.body as string);
+
+    // test that faux response makes it through
+    await expect(hashResponse.structureHash).toEqual("ef7be01282ea97380e4d3bbcba6774cbc7242c46ee51b7e611f1efdfa3623e53");
+
+    // test that fetch is called how validator expects
+    await expect(payload.params[0]?.create_statement).toEqual(createStatement);
```

Comment:
Our tests aren't really testing what the Validator responds with, we would need end-to-end test for that.  My thought here is that this spy will at least ensure that mock response is mapped to the expected SDK's return values.  i.e. snake_case => camelCase.

Reply:
Yeh this is a good addition. I'd really like to have an easy way to actually spin up a server instance and test against "live" response types... that might be something we can get to by end of the next season.

Reply:
That makes sense. I would like to figure out end-to-end tests, but for now I think this works.

Reply:
Excellent!  I've been thinking about end to end, and I don't think it would be that complicated.  Happy to tackle that task, or help out.  I'll admit I'm not exactly sure how that type of task will map to a KR, but I'm sure it will.